chiron/potential.py

```diff
@@ -0,0 +1,96 @@
+import jax
+import jax.numpy as jnp
+from scipy.spatial.distance import pdist, squareform
+from loguru import logger as log
+from openmm import unit
+from openmm.app import Topology
+import jax.numpy as jnp
+
+
+class NeuralNetworkPotential:
+    def __init__(self, model, topology: Topology):
+        self.topology = topology  # The topology of the system
+        if model is None:
+            log.warning("No model provided, using default model")
+        else:
+            self.model = model(self.topology)  # The trained neural network model
+
+    def compute_energy(self, positions):
+        # Compute the pair distances and displacement vectors
+        pair_distances = pdist(positions)
+        displacement_vectors = squareform(pair_distances)
+        # Use the model to compute the potential energy
+        potential_energy = self.model(displacement_vectors)
+        return potential_energy
+
+    def compute_force(self, positions) -> jnp.ndarray:
+        # Compute the force as the negative gradient of the potential energy
+        positions_without_unit = jnp.array(
+            positions.value_in_unit_system(unit.md_unit_system)
+        )
+        force = -jax.grad(self.compute_energy)(positions_without_unit)
+        return force
+
+    def compute_pairlist(self, positions, cutoff) -> jnp.array:
+        # Compute the pairlist for a given set of positions and a cutoff distance
+        pair_distances = pdist(positions)
+        pairlist = jnp.where(pair_distances < cutoff)
+        return pairlist[0]
+
+
+class LJPotential(NeuralNetworkPotential):
+    def __init__(
+        self,
+        topology: Topology,
+        sigma: unit.Quantity = 1.0 * unit.kilocalories_per_mole,
+        epsilon: unit.Quantity = 3.350 * unit.angstroms,
+    ):
+        assert isinstance(sigma, unit.Quantity)
+        assert isinstance(epsilon, unit.Quantity)
+
+        self.sigma = sigma.value_in_unit(
+            unit.kilocalories_per_mole
+        )  # The distance at which the potential is zero
+        self.epsilon = epsilon.value_in_unit(
+            unit.angstrom
+        )  # The depth of the potential well
+        self.topology = topology  # The topology of the system
+
+    def compute_energy(self, positions: unit.Quantity):
+        # Compute the pair distances and displacement vectors
+        positions = jnp.array(positions.value_in_unit(unit.angstrom))
+        pair_distances = pdist(positions)
+        displacement_vectors = squareform(pair_distances)
```

Comment:
The diagonal of `displacement_vectors` is going to be all zeros which will not work in the potential evaluation.